### PR TITLE
OH2-291 | Vaccine list

### PR DESCRIPTION
**Pull request by SilverD3**
See [OH2-291](https://openhospital.atlassian.net/browse/OH2-291)

[OH2-291]: https://openhospital.atlassian.net/browse/OH2-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            return (
+              <Table
+                rowData={formatDataToDisplay(data ?? [])}
+                tableHeader={header}
+                labelData={label}
+                columnsOrder={order}
+                rowsPerPage={10}
+                isCollapsabile={false}
+              />
+            );
```
Comment (SteveGT96) [inline src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx]: Filter is missing. Please take a look at the [mock-up](https://w6nt6m.axshare.com/?code=373a8db2aee9ac436955a60fb179b091&id=ijpxg2&p=vaccines).

**Review by SteveGT96 — CHANGES_REQUESTED** [review]
Unit tests failed

File: src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx (hunk 1 above)
Comment (SteveGT96) [reply src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx]: Please, align with develop

**Comment by gasp** [comment]
> Unit tests failed

It looks like the test failed because of a ECONNRESET. This is usually resolved by re-running the test.

File: src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const header = ["code", "type", "description"];
+
+  const label = {
+    code: t("vaccine.code"),
+    type: t("vaccine.vaccinetype"),
+    description: t("vaccine.description"),
+  };
+
+  const filters: TFilterField[] = [
+    { key: "type", label: t("vaccine.vaccinetype"), type: "text" },
```
Comment (SteveGT96) [inline src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx]: I would be better to handle "type" filter as a select, you could provide the vacines types as options.  You could also refer to existing filters.

**Review by SteveGT96 — CHANGES_REQUESTED** [review]

**Comment by mwithi** [comment]
@SilverD3 please resolve conflicts also (I think it is the topic suggested by @gasp on [Slack](https://join.slack.com/t/openhospitalworkspace/shared_invite/zt-cg9iu1d0-sjTuWWziFgcDqiTWyMJBAw))

File: src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx (hunk 2 above)
Comment (mwithi) [reply src/components/accessories/admin/vaccines/vaccinesTable/VaccinesTable.tsx]: @SilverD3 can you do this?

**Comment by mwithi** [comment]
Is it correct I don't see yet filter values for the type?

**Comment by SilverD3** [comment]
If you're using the backend, maybe it's because of the disability to fetch vaccines types.
With the mock data it's okay.
vaccines (2)

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**